### PR TITLE
Initial Analytics

### PR DESCRIPTION
**Pull request by sephcoster**
Adding analytics module to allow for custom firing of Google Tag Manager events within the application for those events not tracked (or not present on page load) within React.

## Additions

- Analytics.jsx file with event firing logic

## Removals

- None

## Changes

- Updated actions for export to include analytics events

## Testing

- 💯 

## Review

- @JeffreyMFarley 

[Preview this PR without the whitespace changes](?w=0)

## Screenshots
N/A - this is a behind-the-scenes change

## Notes

- None

## Todos

- Continue to add tracking elements where GA does not automatically track elements

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,42 +1,43 @@
-import './App.less';
+import './App.less'
```
Comment (JeffreyMFarley) [inline src/App.jsx]: Get them pesky semicolons out of here

File: src/actions/analytics.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * @name sendEvents
+   * @kind function
+   *
+   * @description
+   * Pushes multiple events to the GTM dataLayer.
+   *
+   * @param {array} eventsArray Array of event objects.
+   */
+  // We don't require sending multiple events at this time.
```
Comment (JeffreyMFarley) [inline src/actions/analytics.jsx]: I'm not crazy about leaving commented-out code in the codebase.  But I do understand the motivation here.

Is there a public repo that has this function already that we could point to?  Something like:

```Javascript
// sendEvents(array) is at http://example.com/foo/master/bar.js
```

File: src/actions/dataExport.jsx [hunk 3]
```diff
@@ -17,6 +18,9 @@ const DATA_HOST = 'https://data.consumerfinance.gov'
 * @returns {string} a packaged payload to be used by Redux reducers
 */
 export function showExportDialog() {
+  Analytics.sendEvent(
+    Analytics.getDataLayerOptions( 'Export', 'User Opens Export Modal' )
+  )
```
Comment (JeffreyMFarley) [inline src/actions/dataExport.jsx]: 👍 👍 👍

File: src/actions/__tests__/analytics.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    Analytics.tagManagerIsLoaded = true
+    Analytics.sendEvent( Analytics.getDataLayerOptions(
+      'woah nelly action',
+      'label',
+    ) )
+    console.log('Window Datalayer: ', window.dataLayer )
+    expect( window.dataLayer.length > 0 ).toBe( true )
+  })
+
+} )
```
Comment (JeffreyMFarley) [inline src/actions/__tests__/analytics.spec.jsx]: I appreciate the dedication to 100% testing!

File: src/actions/analytics.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * @name sendEvent
+   * @kind function
+   *
+   * @description
+   * Pushes an event to the GTM dataLayer.
+   * @param {object} dataLayerOptions Type of event.
+   */
+  sendEvent: function( dataLayerOptions ) {
+    console.log('SEND EVENT GA FUNCTION FIRED')
```
Comment (JeffreyMFarley) [inline src/actions/analytics.jsx]: Can we get rid of the console line?

Comment (JeffreyMFarley) [reply src/actions/analytics.jsx]: Other linting:

```
   1:1   warning  Too many blank lines at the beginning of file. Max of 0 allowed  no-multiple-empty-lines
   2:5   warning  'Analytics' is never reassigned. Use 'const' instead             prefer-const
  59:5   warning  Unexpected console statement                                     no-console
  59:16  warning  There must be a space inside this paren                          space-in-parens
  59:47  warning  There must be a space inside this paren                          space-in-parens
```

Comment (sephcoster) [reply src/actions/analytics.jsx]: The latest updates should take care of the these linting issues - thanks!

**Pull request merged**